### ISSUE TITLE
reset map on restart

### ISSUE THREAD

**Issue by wozniak** [momentum-mod/game#1137]
**What feature is your improvement idea related to? Please describe.**
Some maps give weapons, have you break glass to proceed, etc. When resetting a map it can be easier to get a PB because these things are already done for you.

**Describe the solution you'd like**
clean inventory and replace glass panes, weapons on ground, and other interactive parts of the map

**Describe alternatives you've considered, if any**
Manually reloading the whole map is annoying and takes a while

**Additional context**
this might require engine

**Comment by hexaflexahexagon** [comment]
this is related to the issue of rotating/moving things not being reset on mom_restart, meaning there can be shenanigans where you have to wait for an optimal cycle before starting a run. perhaps all of these issues can be solved map-side instead (as the moving platform issue is on a map like rj_quba already has been)? seems simpler than if it were done through code.

**Comment by wozniak** [comment]
How could you fix broken glass panes? CS:GO does this on round start.

**Comment by Gocnak** [comment]
Maps with one-time breakables tend to be patched to be broken already. Same with locked doors. We may have hooks into rotatables and movables in the future to only start motion when leaving the start zone but the consistency is most important here.

One-off obstacles like these are typically regarded as more annoying than fun to run against, and we'd like to set consistent rules across all game modes with regards to them. So I'll leave this issue open for now, but we may not go down this route of one-time obstacle restoration in the future.

**Comment by wozniak** [comment]
alright. I was just wondering since I've been playing this fun bhop map with one time breakables and weapons lying on the ground
